### PR TITLE
ref: Check options for null before setting on android

### PR DESCRIPTION
**Pull request by denrase**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

- Check all options for nullability before mutating the native Kotlin options.

- We had mostly force casts in Kotlin, which we don't have in iOS integration. I'm assuming the latter is expected behaviour.

- Expose `attachThreads` option with default value `false`. 

- Since `attachStacktrace` also does not have the `is` prefix, i also went for `attachThreads` for the sake of consitency.

**Other**

- `diagnosticLevel` is only set when `options.debug == true` on iOS. Should we mirror this behaviour for the Android integration, or is this a separate issue?

- Could not build the flutter sample locally on Android because of `ndkVersion`. https://github.com/getsentry/sentry-dart/blob/ec19b491f6b2706e8951e33eaa5fcfd3e106dab3/flutter/example/android/app/build.gradle#L69

## :bulb: Motivation and Context

https://github.com/getsentry/sentry-dart/projects/1#card-50176945

## :green_heart: How did you test it?

- Ran the default app for options.
- We don't yet have the means to read native options with the dart sdk, so sadly no tests.
- To verify `attachThreads`, threw a exception on Android Native and observed the thread on sentry.io

Bildschirmfoto 2020-11-28 um 14 41 40

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Review by marandaneto — CHANGES_REQUESTED** [review]
code looks good :)
missing a changelog entry + formatting probably? (see CI checks)

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 1]
```diff
@@ -66,36 +66,57 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
...
-      val sentryLevel = SentryLevel.valueOf(level.toUpperCase(Locale.ROOT))
-      options.setDiagnosticLevel(sentryLevel)
-
-      val anrEnabled = args["anrEnabled"] as Boolean
+      (args["dsn"] as? String)?.let {
+        options.dsn = it
+      }
+      (args["debug"] as? Boolean)?.let {
+        options.isDebug = it
+      }
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: we could move the reading of the `diagnosticLevel` for only when the `debug` flag is enabled, so we save some allocation, good point.

**Comment by marandaneto** [comment]
@denrase 

> Could not build the flutter sample locally on Android because of ndkVersion.

do you mean it did not work at all or you just needed to install it? (the ndkVersion)

**Comment by denrase** [comment]
@marandaneto It worked when i changed the version to the one i have installed in the build.gradle. But good point, i will just go with the one that is referenced. This also removes the risk of commiting a wrong one.

**Review by marandaneto — APPROVED** [review]
left a comment but other than that, LGTM.
Thanks for doing this.

**Comment by denrase** [comment]
Added 'diagnosticLevel' change.

**Comment by denrase** [comment]
Also, the action for too long methods is failing for the kotlin file. should i try to refactor or leave as is?

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 2]
```diff
@@ -66,36 +66,57 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
...
+      (args["maxBreadcrumbs"] as? Int)?.let {
+        options.maxBreadcrumbs = it
+      }
+      (args["cacheDirSize"] as? Int)?.let {
+        options.cacheDirSize = it
+      }
+      (args["diagnosticLevel"] as? String)?.takeUnless { options.isDebug == false }?.let {
+        val sentryLevel = SentryLevel.valueOf(it.toUpperCase(Locale.ROOT))
+        options.setDiagnosticLevel(sentryLevel)
+      }
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: what's about just doing this inside of the:

```
(args["debug"] as? Boolean)?.let {
      options.isDebug = it
      if (options.isDebug){
        // read diagnosticLevel
      }
}
```

in this case, we don't even check if `diagnosticLevel` exists or do `takeUnless`, so we save some allocation.

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt [hunk 3]
```diff
@@ -66,36 +66,57 @@ class SentryFlutterPlugin : FlutterPlugin, MethodCallHandler {
...
+
+      val anrEnabled = (args["anrEnabled"] as? Boolean) ?: options.isAnrEnabled
       options.isAnrEnabled = anrEnabled
 
-      val nativeCrashHandling = args["enableNativeCrashHandling"] as Boolean
+      val nativeCrashHandling = (args["enableNativeCrashHandling"] as? Boolean) ?: false
```
Comment (marandaneto) [inline flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]:
```suggestion
      val nativeCrashHandling = (args["enableNativeCrashHandling"] as? Boolean) ?: true
```

`enableNativeCrashHandling` default value is `true`, so I'd rather fallback to `true`

File: flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt (hunk 2 above)
Comment (denrase) [reply flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: Wouldn't we then be setting it only if `options.debug` is true from the flutter side, and not set it if the native default `options.debug` is set to true? Does this matter?

Comment (marandaneto) [reply flutter/android/src/main/kotlin/io/sentry/flutter/SentryFlutterPlugin.kt]: good catch, it makes sense in both cases :)

**Comment by marandaneto** [comment]
@denrase CI is not happy, probably static analyzers or formatting

**Review by bruno-garcia — APPROVED** [review]

**Review by marandaneto — APPROVED** [review]
LGTM

**Pull request merged**